spinn_front_end_common/utilities/report_functions/reports.py

```diff
@@ -209,10 +210,11 @@ def _write_one_router_partition_report(f, partition):
     routing_infos = FecDataView.get_routing_infos()
     for edge in partition.edges:
         for m_vertex in outgoing:
-            source_placement = FecDataView.get_placement_of_vertex(m_vertex)
+            if isinstance(m_vertex, AbstractVirtual):
+                continue
```

Comment:
I think this means you won't get any routes starting at a virtual vertex, which seems like an ommision...  Note that virtual vertices have keys and masks too I believe...

Reply:
Agree better to report something than nothing.

But this exposes that in [sPyNNaker/unittests/test_sata_connectors/test_sata_2_different_boards_valid_board_address.py](https://github.com/SpiNNakerManchester/sPyNNaker/blob/master/unittests/test_sata_connectors/test_sata_2_different_boards_valid_board_address.py)

The second source ends up with no entry

Which does not happen in 
https://github.com/SpiNNakerManchester/sPyNNaker/blob/master/unittests/test_spinnaker_link_connectors/test_spinnaker_link_2_different_boards_valid_board_address.py